src/main/java/Scheduler.java

```diff
@@ -22,43 +24,121 @@ public Scheduler(ArrayList<Course> courses, ArrayList<Schedule> schedules) {
+            }
+        }
+        return schedule;
+    }
+
+    /** Creates all lecture/tutorial section permutations that pass all filters for one course.
+     *
+     * @param c course that we take lec/tut permutations of
+     * @return all filtered schedules with these lec/tut permutations
+     */
+    private List<Schedule> populatePermutations(Course c) {
+        List<Schedule> populatedSchedules = new ArrayList<>();
+        List<Section> courseLectures = c.getLectures();
+        List<Section> courseTutorials = c.getTutorials();
+
+        for (Section lec : courseLectures) {
+            for(Section tut : courseTutorials) {
+                Schedule tempSchedule = new Schedule();
+                tempSchedule.addLecture(lec);
+                tempSchedule.addTutorial(tut);
+
+                if (this.checkFilters(tempSchedule)) {
+                    populatedSchedules.add(tempSchedule);
                 }
-                if (!newCourse.getTutorials().isEmpty()) {
-                    schedule.addTutorial(newCourse.getTutorials().get(0));
+            }
+        }
+
+        return populatedSchedules;
+    }
+
+    /** Adds creates all permutations of lectures and tutorials that pass filters from a schedule s.
+     *
+     * @param c the course whose lecs/tuts will be permuted
+     * @param s the schedule to which the permutations will be added
+     * @return all added permutations to this schedule
+     */
+    private List<Schedule> extendPermutations(Course c, Schedule s) {
+        List<Schedule> populatedSchedules = new ArrayList<>();
+        List<Section> courseLectures = c.getLectures();
+        List<Section> courseTutorials = c.getTutorials();
+
+        for (Section lec : courseLectures) {
+            for(Section tut : courseTutorials) {
+                Schedule tempSchedule = s.clone();
+                tempSchedule.addLecture(lec);
+                tempSchedule.addTutorial(tut);
+
+                if (this.checkFilters(tempSchedule)) {
+                    populatedSchedules.add(tempSchedule);
                 }
-            } catch (IOException exception) {
-                /**
-                 * In case something goes wrong with the API for a specific course code, we print
-                 * the code and the exception that is thrown.
-                 */
-                System.out.println(
-                        "Exception occured for course "
-                                + courseCode
-                                + " with the following message: \n"
-                                + exception.toString());
             }
         }
+        return populatedSchedules;
+    }
 
-        return schedule;
+
+    /**
+     *
+     * @return true if all filters pass for Schedule given to the method
+     */
+    private boolean checkFilters(Schedule s) {
+        for (Filter f : this.filters) {
+            if (f.checkSchedule(s) == null) {
```

Comment:
I feel like our FIlter behaviour should be changed back to Boolean because this makes genuinely no sense, especially if it's a check function

Reply:
i agree, please talk to siddarth and kenneth